### PR TITLE
AddWindow UX Enhancement

### LINKED ISSUE #59: Include Add Modules field into Add Contact GUI
Extend the functionality of the Add Contact GUI to match that of the CLI add command.

As adding of modules are not a minimal requirement for adding a person, the corresponding text field UI would reflect as optional.

### LINKED ISSUE #82: Unsafe instantiation of AddCommand
image
Should change to private and use a factory method to instantiate an empty `AddCommand`, to prevent unintentional/intentional misuse of an empty `AddCommand` constructor.

### PR DESCRIPTION
**Pull request by bakano98**
It is possible to add modules, and a person's corresponding status.
Furthermore, if a user were to press `ENTER` on any of the last 3 fields (i.e.`Email`, `Status`, or `Modules`), it will submit all the fields inputted in `AddWindow` and execute the logic.
image

~~The logic is not finalised yet and there may still be some changes.~~

~~With this current implementation, a `Person` is added even if the optional fields result in an error (e.g. `status` not being `blacklist` or `favourite`, or a `Module` being of the wrong format).~~

~~A new idea to improve on this would be to prevent anything from being executed if it results in an error; so user has to fix all errors before `AddWindow` executes and closes.~~

~~I will change to the above in the next commit, which will be done somewhere later today.~~

edit 1: Above implementation done with latest commit.

edit 2: Latest commit closes #82 

Closes #59, #70

### CODE REVIEW

File: src/main/java/seedu/address/ui/AddWindow.java [hunk 1]
```diff
@@ -165,19 +188,60 @@ private void handleAdd() {
...
+        // Do not need to handle the fact that the given status might not be valid.
+        if (!status.equals("")) {
+            String setStatus = "status " + lastIndex + " " + status;
+            try {
+                executeCommand(setStatus);
+            } catch (CommandException | ParseException e) {
+                errorLabel.setText("Error encountered\nEnsure status is only blacklist, whitelist, or empty.");
+                return;
+            }
+        }
```
Comment (lawwm) [inline src/main/java/seedu/address/ui/AddWindow.java:212]: Could add SLAP for  handling status and module? Probably unnecessary.

**Review by lawwm — COMMENTED** [review]
LGTM

File: src/main/java/seedu/address/ui/AddWindow.java (hunk 1 above)
Comment (bakano98) [reply src/main/java/seedu/address/ui/AddWindow.java]: Fixed in latest commit, thanks!

**Review by Btaykb — COMMENTED** [review]
LGTM

**Pull request merged**